### ISSUE TITLE
Apple Silicon M1+ native support without Rosetta?

### ISSUE THREAD

**Issue by mailinger-mate** [WCHSoftGroup/ch34xser_macos#6]
Hello, Could you support Apple Silicon natively with M1+ chips?
The current driver requires the Rosetta layer that I would rather avoid.
<img width="250" alt="Screen Shot 2565-07-28 at 13 53 57" src="https://user-images.githubusercontent.com/10153820/181440622-c9370117-5063-4185-81ab-ea7d336380a5.png">

Thank you!

**Comment by moyun-yinyun** [comment]
https://support.apple.com/en-ca/guide/security/secebb113be1/web
https://support.apple.com/en-ca/HT211861

**Comment by moyun-yinyun** [comment]
> Hello, Could you support Apple Silicon natively with M1+ chips? The current driver requires the Rosetta layer that I would rather avoid. <img alt="Screen Shot 2565-07-28 at 13 53 57" width="250" src="https://user-images.githubusercontent.com/10153820/181440622-c9370117-5063-4185-81ab-ea7d336380a5.png">
> 
> Thank you!

https://support.apple.com/en-ca/guide/security/secebb113be1/web
https://support.apple.com/en-ca/HT211861

**Comment by mailinger-mate** [comment]
Thank you, @moyun-yinyun! I am aware of how Rosetta works, and that we can run programs written for the Intel architecture.

My question is if you would consider writing the drivers native for Apple Silicon so we do not need to use Rosetta?
https://isapplesiliconready.com/

**Comment by moyun-yinyun** [comment]
> 

cd /CH34xVCPDriver.app/Contents/MacOS 
file ./CH34xVCPDriver 
./CH34xVCPDriver: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
./CH34xVCPDriver (for architecture x86_64):	Mach-O 64-bit executable x86_64
./CH34xVCPDriver (for architecture arm64):	Mach-O 64-bit executable arm64

**Comment by moyun-yinyun** [comment]
> Thank you, @moyun-yinyun! I am aware of how Rosetta works, and that we can run programs written for the Intel architecture.
> 
> My question is if you would consider writing the drivers native for Apple Silicon so we do not need to use Rosetta? https://isapplesiliconready.com/

maye the CH34xVCPDriver.pkg need rosetta

cd /CH34xVCPDriver.app/Contents/MacOS 
file ./CH34xVCPDriver 
./CH34xVCPDriver: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
./CH34xVCPDriver (for architecture x86_64):	Mach-O 64-bit executable x86_64
./CH34xVCPDriver (for architecture arm64):	Mach-O 64-bit executable arm64
cd CH34xVCPDriver.app/Contents/Library/SystemExtensions/cn.wch.CH34xVCPDriver.dext/_CodeSignature

file ./cn.wch.CH34xVCPDriver 
./cn.wch.CH34xVCPDriver: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
./cn.wch.CH34xVCPDriver (for architecture x86_64):	Mach-O 64-bit executable x86_64
./cn.wch.CH34xVCPDriver (for architecture arm64):	Mach-O 64-bit executable arm64

**Comment by moyun-yinyun** [comment]
> Thank you, @moyun-yinyun! I am aware of how Rosetta works, and that we can run programs written for the Intel architecture.
> 
> My question is if you would consider writing the drivers native for Apple Silicon so we do not need to use Rosetta? https://isapplesiliconready.com/

pleases add me mail lcl@wch.cn

**Comment by mailinger-mate** [comment]
So there is already driver for amr64 but maybe not packaged correctly?

**Comment by mailinger-mate** [comment]
> pleases add me mail [lcl@wch.cn](mailto:lcl@wch.cn)

Do you mean I should send you email about the issue?

**Comment by moyun-yinyun** [comment]
> > pleases add me mail [lcl@wch.cn](mailto:lcl@wch.cn)
> 
> Do you mean I should send you email about the issue?

Yes, I will send you the installation package in dmg format, the PKG installation package must have rosetta installed

**Comment by mailinger-mate** [comment]
Thank you, I have sent you an email

**Comment by andypiper** [comment]
Can we get the Apple Silicon version for everyone please.

**Comment by mailinger-mate** [comment]
I will test the driver this week to confirm if it is working

**Comment by mailinger-mate** [comment]
@moyun-yinyun @andypiper Thank you, I can confirm that the `CH34xVCPDriver.dmg` file you shared with me works well. Would you be able to publish it for everyone as well? 🙏

**Comment by lptr** [comment]
Please publish it for everyone.

**Comment by gingerbeardman** [comment]
There is another way!

1. Get the files out of the .pkg using
  - SuspiciousPackage https://www.mothersruin.com/software/SuspiciousPackage/
  - The Unarchiver https://theunarchiver.com/
2. Move CH34xVCPDriver.app to /Applications
3. Run /Applications/CH34xVCPDriver.app
4. Press Install
5. Reboot

Done.

**Comment by mailinger-mate** [comment]
@moyun-yinyun thank you for sharing the m1 driver with me a few months ago.
Would you please also upload the working DMG installer to this repository for everyone?

**Comment by mela93** [comment]
> There is another way...
> 
>     1. Get the files out of the .pkg using one of:
>        
>        * SuspiciousPackage https://www.mothersruin.com/software/SuspiciousPackage/
>        * The Unarchiver https://theunarchiver.com/
> 
>     2. Move CH34xVCPDriver.app to /Applications manually
> 
>     3. Run /Applications/CH34xVCPDriver.app
> 
>     4. Press Install button
> 
>     5. Reboot
> 
> 
> Done!

Thank you, this is working for me.
